### PR TITLE
Avoid processing duplicate commit messages

### PR DESCRIPTION
**Pull request by vusirikala**
### Description

Making sure the buffer manager processes a commit message from only one validator during sync up. We add `pending_highest_commit_cert` field in round manager that stores highest commit cert that is being synced by the buffer manager during this round. If any sync up request has a lower commit cert than this, we do not send the syncing request to buffer manager again.

### CODE REVIEW

File: consensus/src/round_manager.rs [hunk 1]
```diff
@@ -485,22 +487,42 @@ impl RoundManager {
...
-                        SecurityEvent::InvalidSyncInfoMsg,
-                        sync_info = sync_info,
-                        remote_peer = author,
-                        error = ?e,
-                    );
-                    VerifyError::from(e)
-                })?;
+
+            let highest_commit_round = sync_info.highest_commit_round();
+            let mut sync_commit_certs = false;
```
Comment (sitalkedia) [inline consensus/src/round_manager.rs:492]: @vusirikala - The updated logic looks convoluted and hard to follow. I am wondering if its cleaner if we split up the `sync_up` function into two functions `sync_up_non_commit_cert` and `sync_up_commit_cert`. One thing to take care of is avoid double verification in this and also, we probably need to split the `add_certs` function accordingly. Thoughts? cc - @zekun000

File: consensus/src/round_manager.rs [hunk 2]
```diff
@@ -476,31 +481,83 @@ impl RoundManager {
...
+        let local_sync_info = self.block_store.sync_info();
+        if sync_info.has_newer_commit_certificates(&local_sync_info) {
+            info!(
+                self.new_log(LogEvent::ReceiveNewCertificate)
+                    .remote_peer(author),
+                "Local state {}, remote state {}", local_sync_info, sync_info
+            );
+
+            let highest_commit_round = sync_info.highest_commit_round();
+            if self.pending_highest_commit_round.is_none()
```
Comment (zekun000) [inline consensus/src/round_manager.rs:500]: nit: pending_round.map_or(false, |round| round < highest_commit_round)

File: consensus/src/round_manager.rs [hunk 3]
```diff
@@ -476,31 +481,83 @@ impl RoundManager {
...
+    async fn sync_up_non_commit_certs(
+        &mut self,
+        sync_info: &SyncInfo,
+        author: Author,
+        sync_info_verified: bool,
+    ) -> anyhow::Result<()> {
         let local_sync_info = self.block_store.sync_info();
-        if sync_info.has_newer_certificates(&local_sync_info) {
+        if sync_info.has_newer_non_commit_certificates(&local_sync_info) {
+            // TODO: Duplicate info message. Check if it is necessary.
```
Comment (zekun000) [inline consensus/src/round_manager.rs:535]: we should have separate log entries for this

File: consensus/src/round_manager.rs [hunk 4]
```diff
@@ -525,7 +582,9 @@ impl RoundManager {
         if message_round < self.round_state.current_round() {
             return Ok(false);
         }
-        self.sync_up(sync_info, author).await?;
+        let sync_info_verified = self.sync_up_commit_certs(sync_info, author).await?;
```
Comment (zekun000) [inline consensus/src/round_manager.rs:585]: the var name is a bit confusing, I thought it's a sync info that's already verified, probably should call it `is_sync_info_verified`

File: consensus/src/round_manager.rs [hunk 5]
```diff
@@ -185,6 +185,7 @@ pub struct RoundManager {
     epoch_state: Arc<EpochState>,
     block_store: Arc<BlockStore>,
     round_state: RoundState,
+    pending_highest_commit_round: Option<Round>,
```
[the remark this answers is not in the document]
Comment (vusirikala) [reply consensus/src/round_manager.rs:191]: Done.

File: consensus/src/round_manager.rs (hunk 3 above)
Comment (vusirikala) [reply consensus/src/round_manager.rs:535]: Done.

**Review by sitalkedia — APPROVED** [review]
LGTM. Can you update the test plan with effective latency improvement we are seeing from this?
